ui/src/forms/core/VocabularyField.js

```diff
@@ -32,28 +42,27 @@ export class VocabularyField extends React.Component {
   });
 
   fetchVocabularies = async serializer => {
+    this.cancellableFetchData = withCancel(this.query());
     try {
-      const response = await this.query();
+      const response = await this.cancellableFetchData.promise;
       const options = response.data.hits.map(hit => serializer(hit));
```

Comment:
nit: I find the term `options` a bit misleading since it refers to `response.data.hits`, which are vocabulary entries.